### PR TITLE
Add the AndroidManifest.xml merger

### PR DESCRIPTION
**Pull request by mattleibow**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) 2015 Xamarin
+Copyright (c) .NET Foundation Contributors
```
Comment (mattleibow) [inline LICENSE:3]: Updated this to be the correct one - this is the root license

File: Util/AndroidManifestMerger/External-Dependency-Info.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+All .jar files are under their own licenses.
```
Comment (mattleibow) [inline Util/AndroidManifestMerger/External-Dependency-Info.txt:1]: TODO: add the real licenses

File: Util/AndroidManifestMerger/nuget/Xamarin.Android.ManifestMerger.nuspec [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
+<package>
+  <metadata>
+    <id>Xamarin.Android.ManifestMerger</id>
+    <version>1.0.0</version>
+    <licenseUrl>https://go.microsoft.com/fwlink/?linkid=2102131</licenseUrl>
```
Comment (mattleibow) [inline Util/AndroidManifestMerger/nuget/Xamarin.Android.ManifestMerger.nuspec]: This is a fwlink to the repo root license

File: Util/AndroidManifestMerger/nuget/Xamarin.Android.ManifestMerger.nuspec [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    <authors>Microsoft</authors>
+    <owners>Microsoft</owners>
+    <requireLicenseAcceptance>true</requireLicenseAcceptance>
+    <copyright>© Microsoft Corporation. All rights reserved.</copyright>
+    <tags>Xamarin Xamarin.Android</tags>
+  </metadata>
+  <files>
+    <file src="..\externals\merger\*.jar" target="build/monoandroid" />
+    <file src="..\source\*.targets" target="build/monoandroid" />
+    <file src="..\..\..\LICENSE" target="LICENSE" />
```
Comment (mattleibow) [inline Util/AndroidManifestMerger/nuget/Xamarin.Android.ManifestMerger.nuspec]: Using the repo root license

File: Util/AndroidManifestMerger/build.cake [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+    var settings = new MSBuildSettings()
+        .SetConfiguration("Release")
+        .SetVerbosity(Verbosity.Minimal)
+        .WithRestore();
+
+    MSBuild("./samples/FancyMergingApp.sln", settings);
+
+    // TODO: actually create real tests that build the APK and test that as well
+    //       as ensuring that the target is properly skipped on subsequent builds.
```
Comment (mattleibow) [inline Util/AndroidManifestMerger/build.cake]: Just a TODO for later, we need to add some real tests, not just this smoke test.

**Review by newky2k — APPROVED** [review]

**Pull request merged**